cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowSpecification.java

```diff
@@ -32,21 +34,22 @@
   private final String className;
   private final String name;
   private final String description;
-  private final Map<String, WorkflowActionSpecification> customActionMap;
-  private final List<ScheduleProgramInfo> actions;
   private final Map<String, String> properties;
 
+  private List<WorkflowNode> nodes = Lists.newArrayList();
+  private Map<String, WorkflowActionSpecification> customActionMap = Maps.newHashMap();
```

Comment:
Why we still need this map? Can't it be represented by subclass of `WorkflowNode`? Or you can have `WorkflowActionNode` to be able to carry either `ScheduleProgramInfo` or `WorkflowActionSpecification` for custom action?